src/views/Document.vue

```diff
@@ -118,32 +118,188 @@ export default {
+		},
+		mergeIntervals: function (intervals) {
+			
+			if(intervals.length <= 1)
+				return intervals;
+
+			let stack = [];
+			let top = null;
+
+			intervals = intervals.sort(function (a, b) {
+				if (a.from === b.from){
+					return +(a.to > b.to) || +(a.to === b.to) - 1;
+				}
+				return +(a.from > b.from) || - 1;
+			});
+			
+			intervals[0].matches = [{doc: intervals[0].name, per: intervals[0].percentage}]
+			stack.push(intervals[0]);
+			
+			for (var i = 1; i < intervals.length; i++) {
+				top = stack[stack.length - 1];
+				
+				if (top.to < intervals[i].from) {
+					intervals[i].matches = [{doc: intervals[i].name, per: intervals[i].percentage}]
+					stack.push(intervals[i]);
+				}
+				
+				else if (top.to <= intervals[i].to){
+					top.to = intervals[i].to;
+					top.matches.push({doc: intervals[i].name, per: intervals[i].percentage})
+					stack.pop();
+					stack.push(top);
+				}
+				else if(top.to > intervals[i].to){
+					top.matches.push({doc: intervals[i].name, per: intervals[i].percentage})
+					stack.pop();
+					stack.push(top);
+				}
+			}
+			
+			return stack;
+		},
+		toggleTooltip: function(docs){
+
+			const docs_unique = docs.filter((elem, index) => docs.findIndex(obj => obj.doc === elem.doc) === index)
+			let text = ''
+
+			for (var i = 0; i < docs_unique.length; i++){
+				text = text.concat('Zhoda so súborom: ', docs_unique[i].doc, ' (',  docs_unique[i].per, '%)<br>')
+			}
+			
+			return(text)
+		},
 		highlightedText: function () {
-			const indices = this.document.matched_docs
+			
+			const indices = this.document.result.matched_docs
 				.map((matched_doc) =>
-					matched_doc.matches.map((match) => ({
+					matched_doc.intervals.map((match) => ({
 						name: matched_doc.name,
-						from: match.char_from,
-						to: match.char_to,
+						from: match.begin,
+						to: match.end,
+						percentage: this.$filters.roundToTwoDecimals(this.$filters.toNumber(matched_doc.percentage)),
+						color: this.colorForIndex(this.document.result.matched_docs.map(d => d.name).indexOf(matched_doc.name))
 					}))
 				)
 				.flat();
+		
+			const intervals = this.mergeIntervals(indices)
 
-			indices.sort((a, b) => b.to - b.from - (a.to - a.from));
-
-			const subStringsToReplace = indices.map((h) => ({
+			const subStringsToReplace = intervals.map((h) => ({
 				text: this.document.text.substring(h.from, h.to + 1),
-				doc: h.name,
-			}));
+				docs: h.matches,
+				color: h.color,				
+			}));	
+			
 			return subStringsToReplace.reduce(
 				(string, substring) =>
 					string.replace(
 						new RegExp(substring.text, 'g'),
-						`<span v-tooltip.top="'Zhoda so súborom: ${substring.doc}'" class="text-red-500 font-bold">${substring.text}</span>`
+						`<span v-tooltip.top="'${this.toggleTooltip(substring.docs)}'" class="font-bold" style="color:${substring.color};">${substring.text}</span>`
 					),
 				escape(this.document.text)
 			);
 		},
 	},
 };
 </script>
+
+<style>
+.tooltip {	 
```

Comment:
Can you please add these styles into global styles? Also, please don't use css, we can use scss and apply styles from tailwind.